awx/main/models/credential/__init__.py

```diff
@@ -1171,6 +1171,25 @@ def create(self):
     },
 )
 
+ManagedCredentialType(
+    namespace='gpg_public_key',
+    kind='cryptography',
+    name=gettext_noop('GPG Public Key'),
+    inputs={
+        'fields': [
+            {
+                'id': 'gpg_public_key',
+                'label': gettext_noop('GPG Public Key'),
+                'type': 'string',
+                'secret': True,
+                'multiline': True,
+                'help_text': gettext_noop('GPG Public Key used to validate content signatures.'),
+            },
+        ],
+        'required': ['gpg_public_key'],
+    },
```

Comment:
Related to your other TODO, I would prefer that we add to this:

```python
    injectors={
        'extra_vars': {
            "gpg_pubkey": "{{gpg_public_key}}",
        },
    },
```

Even if it's not used, because at least it articulates what it does. If, for whatever reason, someone used one of these credentials in a job, they could hypothetically employ the related plugins, use the same tasks, and have it work.

Reply:
As we discussed in PM, I'd like to prohibit these from being used in jobs. It feels weird to allow it, like it's just leaking an implementation detail.